contributing.rst

```diff
@@ -1,7 +1,46 @@
-The git stuff
+The Git stuff
 =============
 
-`Guide by github, scroll down a bit <https://guides.github.com/activities/contributing-to-open-source/>`_
+General commit rules
+--------------------
+
+1. All changes introduced by the commit (diff lines) must be related to the subject of
+   the commit.
+
+   If you change some other parts of the file, because your editor reformatted
+   automatically the code, this should be part of a different commit.
+
+   *Tip:* Never commit everything as is using ``git commit -a``, but review carefully
+   your changes with ``git add -p``.
+
+2. Changes should not introduce trailing whitespace.
+
+   Always check your changes for whitespace errors using ``git diff --check`` or add
+   following ``pre-commit`` hook:
+
+   .. code-block:: sh
+
+      #!/bin/sh
+      git diff --check --cached || exit $?
+
+   Nim does not care about trailing whitespace, neither reStructuredText, so you should
```

Comment:
Make that: "No sane programming or markup language cares about trailing whitespace." :P